[BUG] Building against static LibRaw no longer finds static jasper dependency

**Describe the bug**

It looks like the OIIO build stopped being able to properly link against a static LibRaw and all of its dependencies between 2.2.15.0 and 2.2.16.0.

I'm building against a static build of LibRaw 0.20.2 (which was built with `--enable-jpeg --enable-jasper --enable-lcms`), and in OIIO 2.2.15.0, the static `jasper` dependency is resolved and linked appropriately:

```
-- Found LibRaw: /luma/soft/rez/packages/libraw/0.20.2/platform-linux/gcc-6.3.1/lib/libraw.a (found version "0.20.2") 
-- Found JPEG: /luma/soft/rez/packages/jpeg/9.c/platform-linux/gcc-6.3.1/lib/libjpeg.a (found version "90") 
-- Found Jasper: /luma/soft/rez/packages/jasper/2.0.16/platform-linux/gcc-6.3.1/lib64/libjasper.a (found version "2.0.16") 
-- Found LibRaw 0.20.2 
--     LibRaw_INCLUDE_DIR = /luma/soft/rez/packages/libraw/0.20.2/platform-linux/gcc-6.3.1/include
--     LibRaw_LIBRARIES = /luma/soft/rez/packages/libraw/0.20.2/platform-linux/gcc-6.3.1/lib/libraw.a
--     LibRaw_r_LIBRARIES = /luma/soft/rez/packages/libraw/0.20.2/platform-linux/gcc-6.3.1/lib/libraw_r.a;/luma/soft/rez/packages/jasper/2.0.16/platform-linux/gcc-6.3.1/lib64/libjasper.a;/luma/soft/rez/packages/jpeg/9.c/platform-linux/gcc-6.3.1/lib/libjpeg.a;/luma/soft/rez/packages/lcms/2.9/platform-linux/gcc-6.3.1/lib/liblcms2.a
```

However, building 2.2.16.0+ with the same dependencies/build configuration no longer resolves all of LibRaw's second-order static dependencies:

```
-- Found LibRaw: /luma/soft/rez/packages/libraw/0.20.2/platform-linux/gcc-6.3.1/lib/libraw.a (found version "0.20.2") 
-- Found LibRaw 0.20.2 
--     LibRaw_INCLUDE_DIR = /luma/soft/rez/packages/libraw/0.20.2/platform-linux/gcc-6.3.1/include
--     LibRaw_LIBRARIES = /luma/soft/rez/packages/libraw/0.20.2/platform-linux/gcc-6.3.1/lib/libraw.a
--     LibRaw_r_LIBRARIES = /luma/soft/rez/packages/libraw/0.20.2/platform-linux/gcc-6.3.1/lib/libraw_r.a;/luma/soft/rez/packages/lcms/2.9/platform-linux/gcc-6.3.1/lib/liblcms2.a
```

This results in link errors for anything trying to use `libOpenImageIO`, e.g:
```
[ 83%] Linking CXX executable ../../bin/idiff
../../lib/libOpenImageIO.so.2.2.16: undefined reference to `jas_stream_fdopen'
../../lib/libOpenImageIO.so.2.2.16: undefined reference to `jas_image_destroy'
../../lib/libOpenImageIO.so.2.2.16: undefined reference to `jas_stream_close'
../../lib/libOpenImageIO.so.2.2.16: undefined reference to `jas_image_readcmpt'
../../lib/libOpenImageIO.so.2.2.16: undefined reference to `jas_image_decode'
../../lib/libOpenImageIO.so.2.2.16: undefined reference to `jas_stream_fopen'
../../lib/libOpenImageIO.so.2.2.16: undefined reference to `jas_stream_memopen'
../../lib/libOpenImageIO.so.2.2.16: undefined reference to `jas_matrix_create'
../../lib/libOpenImageIO.so.2.2.16: undefined reference to `jas_init'
../../lib/libOpenImageIO.so.2.2.16: undefined reference to `jas_matrix_destroy'
../../lib/libOpenImageIO.so.2.2.16: undefined reference to `jas_stream_seek'
collect2: error: ld returned 1 exit status
```

From looking through the commit history, I'm fairly certain the culprit is in [8ede4690927e3074dab379379ac90e9547fdfb71](https://github.com/OpenImageIO/oiio/commit/8ede4690927e3074dab379379ac90e9547fdfb71#diff-c470226216e28a3919497ddd3194dcda83426b8edb884e62648be47d1b99d36e).

It's somewhat unclear why the search for Jasper was removed... are you able to provide any more context here?

Thanks.

--- a/src/cmake/modules/FindLibRaw.cmake
+++ b/src/cmake/modules/FindLibRaw.cmake
@@ -86,6 +86,10 @@ MARK_AS_ADVANCED(LibRaw_VERSION_STRING
 
 if (LINKSTATIC)
     # Necessary?
+    find_package (Jasper)
+    if (JASPER_FOUND)
+        set (LibRaw_r_LIBRARIES ${LibRaw_r_LIBRARIES} ${JASPER_LIBRARIES})
+    endif()
     find_library (LCMS2_LIBRARIES NAMES lcms2)
     if (LCMS2_LIBRARIES)
         set (LibRaw_r_LIBRARIES ${LibRaw_r_LIBRARIES} ${LCMS2_LIBRARIES})

